json/contextual/src/fs2/data/json/SelectorInterpolator.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2019 Lucas Satabin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package fs2
+package data
+package json
+
+import cats.implicits._
+
+import contextual._
+
+import scala.util.{Failure, Success, Try}
+
+object SelectorInterpolator extends Verifier[Selector] {
+
+  def check(string: String): Either[(Int, String),Selector] =
+    new SelectorParser[Try](string).parse() match {
```

Comment:
Why `Try` if you're returning `Either`? `Either` and then `leftMap` or `adaptError` would seem more idiomatic to me...

Reply:
Yeah I notice before seeing your comment and pushed a version with `Either`.